src/mpi/coll/ibcast/ibcast_tsp_tree.c

```diff
@@ -50,19 +51,38 @@ int MPIR_TSP_Ibcast_sched_intra_tree(void *buffer, MPI_Aint count, MPI_Datatype
      * correctly with pipelined broadcast, for example, buffer+offset
      * if being calculated correctly */
     for (i = 0; i < num_chunks; i++) {
+#ifdef HAVE_ERROR_CHECKING
+        struct MPII_Ibcast_state *ibcast_state = NULL;
+#endif
         int msgsize = (i == 0) ? chunk_size_floor : chunk_size_ceil;
 
+#ifdef HAVE_ERROR_CHECKING
+        ibcast_state = MPIR_TSP_sched_malloc(sizeof(struct MPII_Ibcast_state), sched);
+        if (ibcast_state == NULL)
+            MPIR_ERR_POP(mpi_errno);
+        ibcast_state->n_bytes = msgsize * type_size;
+#endif
+
         /* For correctness, transport based collectives need to get the
          * tag from the same pool as schedule based collectives */
         mpi_errno = MPIR_Sched_next_tag(comm, &tag);
         MPIR_ERR_CHECK(mpi_errno);
 
         /* Receive message from parent */
         if (my_tree.parent != -1) {
+#ifdef HAVE_ERROR_CHECKING
+            mpi_errno =
+                MPIR_TSP_sched_irecv_status((char *) buffer + offset * extent, msgsize,
+                                            datatype, my_tree.parent, tag, comm,
+                                            &ibcast_state->status, sched, 0, NULL, &recv_id);
+            MPIR_ERR_COLL_CHECKANDCONT(mpi_errno, errflag);
+            MPIR_TSP_sched_cb(&MPII_Ibcast_sched_test_length, ibcast_state, sched, 1, &recv_id);
+#else
             mpi_errno =
                 MPIR_TSP_sched_irecv((char *) buffer + offset * extent, msgsize, datatype,
                                      my_tree.parent, tag, comm, sched, 0, NULL, &recv_id);
             MPIR_ERR_COLL_CHECKANDCONT(mpi_errno, errflag);
+#endif
```

Comment:
If `MPIR_TSP_sched_irecv_status` supports `MPI_STATUS_IGNORE`, is there any reason not to always use it?

Reply:
Maybe I missed something, but for ibcast, user does not pass a status argument.

Reply:
if no error checking, set `ibcast_state->status = MPI_STATUS_IGNORE`.

Reply:
maybe something like this
```c
            MPI_Status *status = MPI_STATUS_IGNORE;
#ifdef HAVE_ERROR_CHECKING
            status = &ibcast_state->status;
#endif
            mpi_errno =
                MPIR_TSP_sched_irecv_status((char *) buffer + offset * extent, msgsize,
                                            datatype, my_tree.parent, tag, comm,
                                            status, sched, 0, NULL, &recv_id);
            MPIR_ERR_COLL_CHECKANDCONT(mpi_errno, errflag);
            MPIR_TSP_sched_cb(&MPII_Ibcast_sched_test_length, ibcast_state, sched, 1, &recv_id);
```

Reply:
I see, in that case, we don't need to have two separate vertices (sched_irecv and sched_irecv_status). They can be combined into one after making it supporting MPI_STATUS_IGNORE.

This sounds good to me. But can we do that in a separate PR?

Reply:
> This sounds good to me. But can we do that in a separate PR?

Sure.